ldap/src/main/java/org/springframework/security/ldap/server/UnboundIdContainer.java

```diff
@@ -108,23 +112,40 @@ public void start() {
 			this.running = true;
 		} catch (LDAPException ex) {
 			throw new RuntimeException("Server startup failed", ex);
+		} catch (IllegalArgumentException ex) {
+			throw ex;
 		}
 
 	}
 
-	private void importLdif(InMemoryDirectoryServer directoryServer) {
+	private void importLdif(InMemoryDirectoryServer directoryServer) throws LDAPException {
 		if (StringUtils.hasText(this.ldif)) {
 			try {
-				Resource[] resources = this.context.getResources(this.ldif);
-				if (resources.length > 0 && resources[0].exists()) {
-					try (InputStream inputStream = resources[0].getInputStream()) {
-						directoryServer.importFromLDIF(false, new LDIFReader(inputStream));
-					}
+
+				Resource resource = locateResource();
+				try (InputStream inputStream = resource.getInputStream()) {
+					directoryServer.importFromLDIF(false, new LDIFReader(inputStream));
 				}
 			} catch (Exception ex) {
 				throw new IllegalStateException("Unable to load LDIF " + this.ldif, ex);
 			}
 		}
+
+	}
+
+	private Resource locateResource() throws IOException , NullPointerException {
+		Resource[] resources = this.context.getResources(this.ldif);
+		if (null == resources || 0 == resources.length) {
+			throw new IllegalArgumentException("requested resource is not found");
+		}
+		return Arrays.stream(resources).filter(resource ->
```

Comment:
In order to reduce the overhead of creating additional objects, we prefer to use For Loops instead of streams.
For more information see gh-7154.